### DISCUSSION TITLE
Invalid UTF-8 string in BSON document

### DISCUSSION THREAD

**Discussion by mkagit** [danny-avila/LibreChat#4082] [Q&A]
### What happened?

I'm getting this errors after the latest buildings. Looks likes it's a utf-8 string issue with Node. 

I'm using MongoDB Atlas

In this thread below they're saying it's something related to Node update.

[[](https://www.mongodb.com/community/forums/t/error-invalid-utf-8-string-in-bson-document-after-rebuild-on-windows/293410)](https://www.mongodb.com/community/forums/t/error-invalid-utf-8-string-in-bson-document-after-rebuild-on-windows/293410/2)

[](https://medium.com/@fs-doc/dealing-with-invalid-utf-8-string-in-bson-document-error-in-mongodb-27f0d2f5c114
)
### Steps to Reproduce

1. Building latest dev
2. Using MongoDB Atlas
3. Meilisearch logging errors.

### What browsers are you seeing the problem on?

Microsoft Edge

### Relevant log output

```shell
`syncWithMeili] Error adding document to Meili Invalid UTF-8 string in BSON document
`
`MeiliMongooseModel.deleteMany] There was an issue deleting conversation indexes upon deletion, next startup may be slow due to syncing Invalid UTF-8 string in BSON document`
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mkagit** [comment]
https://medium.com/@fs-doc/dealing-with-invalid-utf-8-string-in-bson-document-error-in-mongodb-27f0d2f5c114

**Comment by danny-avila** [comment]
Not sure what the issue is exactly? Not having issues indexing conversations/messages, also using meilisearch + Atlas, maybe you don't have meilisearch running?

**Reply by mkagit** [reply]
It happens when npm run with the latest local Node version. Please, read the article from the link above. 
It corrupted my DB and had to delete all chats.

**Reply by danny-avila** [reply]
Seems like a node issue. I'm using node v20 locally `v20.15.1`, the docker images use latest node v20 and not node v22

**Comment by mkagit** [comment]
Yes, I see. I just nvm un-installed 22.7 and installed 20.17. But the
damage has been already done. I had to erase all chats.

Here, he explains what happened:

_But how can this happen?

Well, the vast majority of the time this is due to improper validation of
the data you insert into the database and you will need to check your code
to make sure you are inserting valid UTF-8 characters.

But… it can also happen suddenly because one of your dependencies starts
doing strange things with strings after an automatic or unintentional
update.

This happened recently with Node.js version 22.7.0, which had major bugs in
the UTF-8 validation of strings and inserted invalid characters into the
strings, which were then passed to the database, corrupting it. Suddenly,
code that was working perfectly, started to mess up._
